include/flatbuffers/base.h

```diff
@@ -233,12 +233,17 @@ namespace flatbuffers {
       }
       #define FLATBUFFERS_HAS_STRING_VIEW 1
     // Check for absl::string_view
-    #elif __has_include("absl/strings/string_view.h") && (__cplusplus >= 201411)
-      #include "absl/strings/string_view.h"
-      namespace flatbuffers {
-        typedef absl::string_view string_view;
-      }
-      #define FLATBUFFERS_HAS_STRING_VIEW 1
+    #elif __has_include("absl/strings/string_view.h") && \
+          __has_include("absl/base/config.h") && \
+          (__cplusplus >= 201411)
+      #include "absl/base/config.h"
+      #if !defined(ABSL_USES_STD_STRING_VIEW)
```

Comment:
I don't get this logic, if ABSL does alias to std::string_view, won't this disable string view for flatbuffers?

Reply:
No, std::string_view comes with C++17 in the standard. Flatbuffers already uses std::string_view if compiled with C++17 and later versions. The line 222

```cpp
    #if __has_include(<string_view>) && (__cplusplus >= 201606 || (defined(_HAS_CXX17) && _HAS_CXX17))
      #include <string_view>
      namespace flatbuffers {
        typedef std::string_view string_view;
      }
      #define FLATBUFFERS_HAS_STRING_VIEW 1
```

So it does not rely on the absl alias. absl::string_view should be used only if C++ version is 14 or absl is compiled without `ABSL_USES_STD_STRING_VIEW` flag. Otherwise(C++17 and later versions are available), flatbuffers already choose to run with std::string_view.

Let me know if you have more questions on this.

Reply:
Thanks, that clarifies it.